src/components/common/Forms/EditCreditCardForm/EditCreditCardForm.tsx

```diff
@@ -0,0 +1,198 @@
+import styles from "./EditCreditCardForm.module.scss";
+import * as React from "react";
+import cn from "classnames";
+import Input from "components/common/Input/Input";
+import * as yup from "yup";
+import { extractErrors } from "utils/forms";
+
+const editCreditCardSchema = yup.object().shape({
+  creditCardNumber: yup.string().required("Required"),
+  creditCardName: yup.string().required("Required"),
+  cardDate: yup
+    .string()
+    .required("Required")
+    .matches(/^(0[1-9]|1[0-2])\/?([0-9]{4}|[0-9]{2})$/, "Invalid date"),
+  cardCVV: yup
+    .string()
+    .required("Required")
+    .matches(/^[0-9]{3,4}$/, "Invalid CVV"),
+});
+
+export const DEFAULT_CREDIT_CARD_FORM_VALUES: CreditCardFormValuesT = {
+  id: null,
+  creditCardNumber: "",
+  creditCardName: "",
+  cardDate: "",
+  cardCVV: "",
+};
+
+export type CreditCardFormValuesT = {
+  id: string;
+  creditCardNumber: string;
+  creditCardName: string;
+  cardDate: string;
+  cardCVV: string;
+};
+
+type CreditCardFormErrorsT = {
+  creditCardNumber: string | null;
+  creditCardName: string | null;
+  cardDate: string | null;
+  cardCVV: string | null;
+};
+
+type Props = {
+  initialValues?: CreditCardFormValuesT;
+  onSave?: (creditCardValues: CreditCardFormValuesT) => void;
+  onCancel?: (id: string) => void;
+  onDelete?: (id: string) => void;
+  visible?: boolean;
+};
+
+type State = {
+  values: CreditCardFormValuesT;
+  errors: CreditCardFormErrorsT;
+  editMode: boolean;
+};
+
+export default class EditCreditCardForm extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      values: props.initialValues || DEFAULT_CREDIT_CARD_FORM_VALUES,
+      editMode: !!this.props.initialValues,
+      errors: {
+        creditCardNumber: null,
+        creditCardName: null,
+        cardDate: null,
+        cardCVV: null,
+      },
+    };
+  }
+
+  render() {
+    return (
+      <div
+        className={cn(styles.EditCreditCardForm, {
+          [styles.visible]: this.props.visible === true,
+        })}
+      >
+        <div className={styles.title}>
+          {this.state.editMode ? "Edit this Card" : "Add a New Card"}
+        </div>
+        <div className={styles.form}>
+          <div className={styles.cardNumber}>
+            <div className={styles.inputHint}>Card Number</div>
+            <Input
```

Comment:
We have some examples of more specialized inputs for the credit card number / expiration date that automatically format the text in the input. See `src/components/common/Forms/CreditCardForm`. Relevant props: `formatter`, `parser`, and `inputMode`